src/components/SyncView.js

```diff
@@ -26,50 +27,69 @@ var messages = {
 export default class SyncView extends React.Component {
   constructor (props) {
     super(props)
-    var self = this
     var ready = 'replication-ready'
     this.state = {
       message: messages[ready],
-      status: ready
+      status: ready,
+      targets: []
     }
-
-    ipcRenderer.on('select-file', function (event, file) {
-      if (!file) return
-      self.selectFile(file)
-    })
   }
 
   onClose () {
     this.props.changeView(MapEditor)
   }
 
-  selectFile (filename) {
+  onError (err) {
+    this.setState({
+      message: 'Error: ' + err.message,
+      status: 'replication-error'
+    })
+  }
+
+  replicate (target) {
     var self = this
-    replicate.start(filename, function (err, res, body) {
+    if (!target) return
+    this.destroyed = false
+    this.stream = replicate.start(target)
+    this.stream.on('data', function (data) {
```

Comment:
Be wary here! Setting up a `data` listener uncorks the stream and data begins flowing immediately. Depending on how the replication protocol is implemented, data could be lost before `this.stream` ever gets piped into something.

Reply:
so better to use a through2 stream?

Reply:
+1 that'd work!